javalib/src/main/scala/java/net/URI.scala

```diff
@@ -1,33 +1,1415 @@
+      }
+      if (fragment != null) {
+        uri.append('#')
+        uri.append(quoteComponent(fragment, allLegal))
+      }
+      new Helper().parseURI(uri.toString, true)
+    }
+  }
+
+  /**
+   * Creates a new URI instance using the given arguments. This constructor
+   * first creates a temporary URI string from the given components. This
+   * string will be parsed later on to create the URI instance.
+   * <p>
+   * {@code [scheme:]host[path][#fragment]}
+   *
+   * @param scheme
+   *            the scheme part of the URI.
+   * @param host
+   *            the host name of the URI.
+   * @param path
+   *            the path to the resource on the host.
+   * @param fragment
+   *            the fragment part of the URI.
+   * @throws URISyntaxException
+   *             if the temporary created string doesn't fit to the
+   *             specification RFC2396 or could not be parsed correctly.
+   */
+  def this(scheme: String, host: String, path: String, fragment: String) =
+    this(scheme, null, host, -1, path, null, fragment)
+
+  /**
+   * Creates a new URI instance using the given arguments. This constructor
+   * first creates a temporary URI string from the given components. This
+   * string will be parsed later on to create the URI instance.
+   * <p>
+   * {@code [scheme:][//authority][path][?query][#fragment]}
+   *
+   * @param scheme
+   *            the scheme part of the URI.
+   * @param authority
+   *            the authority part of the URI.
+   * @param path
+   *            the path to the resource on the host.
+   * @param query
+   *            the query part of the URI to specify parameters for the
+   *            resource.
+   * @param fragment
+   *            the fragment part of the URI.
+   * @throws URISyntaxException
+   *             if the temporary created string doesn't fit to the
+   *             specification RFC2396 or could not be parsed correctly.
+   */
+  def this(scheme: String,
+           authority: String,
+           path: String,
+           query: String,
+           fragment: String) = {
+    this()
+    if (scheme != null && path != null && path.length > 0 && path.charAt(0) != '/') {
+      throw new URISyntaxException(path, "Relative path")
+    }
+    val uri: StringBuilder = new StringBuilder()
+    if (scheme != null) {
+      uri.append(scheme)
+      uri.append(':')
+    }
+    if (authority != null) {
+      uri.append("//")
+      uri.append(quoteComponent(authority, "@[]" + someLegal))
+    }
+    if (path != null) {
+      uri.append(quoteComponent(path, "/@" + someLegal))
+    }
+    if (query != null) {
+      uri.append('?')
+      uri.append(quoteComponent(query, allLegal))
+    }
+    if (fragment != null) {
+      uri.append('#')
+      uri.append(quoteComponent(fragment, allLegal))
+    }
+    new Helper().parseURI(uri.toString, false)
+  }
+
+  private class Helper {
+
+    @throws(classOf[URISyntaxException])
```

Comment:
The `@throws` annotations are only affecting the scaladoc, so you should get rid of them too.